packages/webview_flutter/webview_flutter_platform_interface/lib/src/v4/types/web_resource_error_delegate.dart

```diff
@@ -0,0 +1,160 @@
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+
+import '../webview_platform.dart';
+
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// Possible error type categorizations used by [WebResourceError].
+enum WebResourceErrorType {
+  /// User authentication failed on server.
+  authentication,
+
+  /// Malformed URL.
+  badUrl,
+
+  /// Failed to connect to the server.
+  connect,
+
+  /// Failed to perform SSL handshake.
+  failedSslHandshake,
+
+  /// Generic file error.
+  file,
+
+  /// File not found.
+  fileNotFound,
+
+  /// Server or proxy hostname lookup failed.
+  hostLookup,
+
+  /// Failed to read or write to the server.
+  io,
+
+  /// User authentication failed on proxy.
+  proxyAuthentication,
+
+  /// Too many redirects.
+  redirectLoop,
+
+  /// Connection timed out.
+  timeout,
+
+  /// Too many requests during this load.
+  tooManyRequests,
+
+  /// Generic error.
+  unknown,
+
+  /// Resource load was canceled by Safe Browsing.
+  unsafeResource,
+
+  /// Unsupported authentication scheme (not basic or digest).
+  unsupportedAuthScheme,
+
+  /// Unsupported URI scheme.
+  unsupportedScheme,
+
+  /// The web content process was terminated.
+  webContentProcessTerminated,
+
+  /// The web view was invalidated.
+  webViewInvalidated,
+
+  /// A JavaScript exception occurred.
+  javaScriptExceptionOccurred,
+
+  /// The result of JavaScript execution could not be returned.
+  javaScriptResultTypeIsUnsupported,
+}
+
+/// Error returned in `WebView.onWebResourceError` when a web resource loading
+/// error has occurred.
+///
+/// Platform specific implementations can add additional fields by extending
+/// this class and provide a factory method that takes the
+/// [WebResourceErrorDelegate] as a parameter.
+///
+/// {@tool sample}
+/// This example demonstrates how to extend the [WebResourceErrorDelegate] to
+/// provide additional platform specific parameters.
+///
+/// Note that the additional parameters should always accept `null` or have a
+/// default value to prevent breaking changes.
+///
+/// ```dart
+/// class IOSWebResourceError extends WebResourceErrorDelegate {
+///   IOSWebResourceError._(
+///     WebResourceErrorDelegate webResourceError,
+///     this.domain,
+///   ) : super(
+///     errorCode: webResourceError.errorCode,
+///     description: webResourceError.description,
+///     errorType: webResourceError.errorType,
+///   );
+///
+///   factory IOSWebResourceError.fromWebResourceError(
+///     WebResourceErrorDelegate webResourceError, {
+///     String? domain,
+///   }) {
+///     return IOSWebResourceError._(
+///       webResourceError: webResourceError,
+///       domain: domain,
+///     );
+///   }
+///
+///   final String? domain;
+/// }
+/// ```
+/// {@end-tool}
+class WebResourceErrorDelegate extends PlatformInterface {
```

Comment:
This doesn't need to be a delegate since it is instantiated by the platform and returned to the user. It's similar to `JavaScriptMessage`. The documentation above is great though!